airflow-core/tests/unit/models/test_deadline.py

```diff
@@ -210,13 +211,46 @@ def test_handle_miss_async_callback(self, dagrun, session):
 
     @pytest.mark.db_test
     @pytest.mark.parametrize(
-        "event, none_trigger_id_expected",
+        "event, none_trigger_expected, expected_state",
         [
-            pytest.param(TriggerEvent({"status": "success"}), True, id="success_event"),
-            pytest.param(TriggerEvent({"status": "some status"}), False, id="unknown_event"),
+            pytest.param(
+                TriggerEvent({EVENT_STATE_KEY: DeadlineCallbackState.SUCCESS, "result": "test_result"}),
+                True,
+                DeadlineCallbackState.SUCCESS,
+                id="success_event",
+            ),
+            pytest.param(
+                TriggerEvent({EVENT_STATE_KEY: DeadlineCallbackState.STARTED}),
+                False,
+                DeadlineCallbackState.STARTED,
+                id="started_event",
+            ),
+            pytest.param(
+                TriggerEvent(
+                    {EVENT_STATE_KEY: DeadlineCallbackState.NOT_FOUND, "message": "module not found"}
+                ),
+                True,
+                DeadlineCallbackState.NOT_FOUND,
+                id="not_found_event",
+            ),
+            pytest.param(
+                TriggerEvent({EVENT_STATE_KEY: DeadlineCallbackState.NOT_AWAITABLE}),
+                True,
+                DeadlineCallbackState.NOT_AWAITABLE,
+                id="not_awaitable_event",
+            ),
+            pytest.param(
+                TriggerEvent(
+                    {EVENT_STATE_KEY: DeadlineCallbackState.OTHER_FAILURE, "message": "error message"}
+                ),
+                True,
+                DeadlineCallbackState.OTHER_FAILURE,
+                id="failed_event",
+            ),
+            pytest.param(TriggerEvent({EVENT_STATE_KEY: "unknown_state"}), False, None, id="unknown_event"),
         ],
```

Comment:
Any particular reason not to just pull `expected_state` from `event`?  It looks like every case being tested, the expected state is already right there in the event.  First line of the test could just be `expected_state = event[EVENT_STATE_KEY]` and save a bunch of redundancy, no?

Reply:
It was all except for 1 case but I've changed it to not have `expected_state` as parameter